aidants_connect_web/templates/aidants_connect_web/users_search_bar.html

```diff
@@ -0,0 +1,14 @@
+{% comment %}
+This search bar is designed to be used with JS logic in
+aidants_connect_web/static/js/users_search.js
+
+Just add it anywhere you display users along with the following static files:
+- aidants_connect_web/static/css/users_search.css
+- aidants_connect_web/static/js/users_search.js
+{% endcomment %}
+<div class="user-search row" hidden>
+  <label>
```

Comment:
Attention, inclure l'input dans son label n'est pas valide pour l'accessibilité, même si çaMarche™ à la souris. Il faut bien un id et un for.

Reply:
(Tu n'es pas obligé de sortir l'input du label, mais il faut bien ajouter le for sur le label)

Reply:
Si, si, ça marche. La version avec le `for` est pour les cas plus complexes où le label et son contrôle ne peuvent être directement liés. Par exemple, ils sont placés dans des `<div>` pour y mettre de l'animation.

Extrait de [la doc MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#technical_summary):

> Content categories : Flow content, phrasing content, interactive content, form-associated element, palpable content.

avec *form-associated element* qui peut être :

- `<button>`
- `<fieldset>`
- `<input>`
- `<keygen>`
- `<label>`
- `<meter>`
- `<object>`
- `<output>`
- `<progress>`
- `<select>`
- `<textarea>`

Reply:
Ça marche mais c'est pas conforme (ça ne passe pas un audit RGAA) et la compatibilité avec les techno d'assistance est incertaine.

Reply:
Cf. Les tests du critère 11.1 du RGAA https://www.numerique.gouv.fr/publications/rgaa-accessibilite/methode-rgaa/criteres/#crit-11-1
Le RGAA fournit un moyen simple d'être, sur les cas simples comme ici, à peu près sûr qu'on va être compatible avec la plupart des lecteurs d'écrans et autres techno d'assistance. Et cette technique de input inclus dans le label a été explicitement signalée en formation comme étant non conforme parce que sur certains lecteurs d'écran, le lien label-input n'est pas fait.

Reply:
Mais dafuq... C'est standard depuis l'aube des temps :expressionless: 
Bon ben je change alors.

Reply:
C'est pas parce que le test du "je clique sur le label et ça met le focus dans le champ" passe bien que c'est standard ou accessible. Pis bon c'est pas non plus comme s'il fallait réécrire toute la page. :)
Je suis allée remonter jusqu'au WCAG et c'est pareil, ça n'est pas une technique conforme : https://www.w3.org/WAI/WCAG21/Techniques/html/H44.html

Et d'ailleurs la même page du WGAC confirme que le <input type="submit">` n'est pas censé prendre de `<label for="">`, je suis curieuse de savoir ce qui est pris en compte dans les technos d'assistance. (Je regarderai demain)